File: synapse/examples/typescript/src/jeff-agent.ts
// src/jeff-agent.ts — Jeff discovers Bob and sends a test message
import Synapse from "./synapse.js";

async function main() {
  const mesh = await Synapse.connect("nats://localhost:4222");

  await mesh.register({
    name: "Jeff's Agent",
    capabilities: [],
    skills: [],
  });

  console.log("Jeff agent online, discovering Bob...");

  const agents = await mesh.discover({ capabilities: ["chat"] });
  const bob = agents.find((a) => a.name === "Bob's Agent");

  if (!bob) {
    console.log("Could not find Bob! Is bob-agent.ts running?");
    await mesh.close();
    return;
  }

  console.log(`Found Bob: ${bob.name} (${bob.id})`);

  const response = await mesh.request(bob.id, "chat", { text: "Hey Bob, how's it going?" });
  console.log(`Bob's response: ${JSON.stringify(response.payload.output)}`);

  await mesh.close();
}

main().catch(console.error);

File: synapse/examples/go/jeff_agent.go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/synapse-go/examples/synapse"
)

func main() {
	mesh, err := synapse.Connect("nats://localhost:4222")
	if err != nil {
		log.Fatalf("Failed to connect: %v", err)
	}
	defer mesh.Close()

	mesh.Register("Jeff's Agent", "", nil, nil)

	agents, err := mesh.Discover([]string{"chat"}, 2*time.Second)
	if err != nil || len(agents) == 0 {
		log.Fatal("Could not find Bob!")
	}

	bob := agents[0]
	fmt.Printf("Found Bob: %s\n", bob.ID)

	response, err := mesh.Request(bob.ID, "chat", map[string]interface{}{
		"text": "Hey Bob, how's it going?",
	}, 30*time.Second)

	if err != nil {
		log.Fatalf("Request failed: %v", err)
	}

	fmt.Printf("Bob's response: %v\n", response.Payload)
}
